tests/transforms/test_metric_tensor.py

```diff
@@ -618,119 +594,528 @@ def test_multi_qubit_gates(self):
+    qml.RZ(fixed_pars[0], wires=[1])
+    qml.CNOT(wires=[0, 1])
+    qml.RX(params0, wires=[0])
+    qml.RX(params0, wires=[1])
+    qml.CNOT(wires=[0, 1])
+    qml.RY(fixed_pars[4], wires=[1])
+    qml.RY(params1, wires=[0])
+    qml.RY(params1, wires=[1])
+    qml.CNOT(wires=[0, 1])
+    qml.RX(fixed_pars[2], wires=[1])
+
+
+fubini_ansatze = [
+    fubini_ansatz0,
+    fubini_ansatz1,
+    fubini_ansatz2,
+    fubini_ansatz3,
+    fubini_ansatz4,
+    fubini_ansatz5,
+    fubini_ansatz6,
+    fubini_ansatz7,
+    fubini_ansatz8,
+]
+
+fubini_params = [
+    (np.array([0.3434, -0.7245345], requires_grad=True),),
+    (
+        np.reshape(
+            [
+                0.73,
+                0.49,
+                0.04,
+                0.29,
+                0.45,
+                0.59,
+                0.64,
+                0.06,
+                0.26,
+                0.93,
+                0.14,
+                0.46,
+                0.31,
+                0.83,
+                0.79,
+                0.25,
+                0.40,
+                0.16,
+            ],
+            (2, 3, 3),
+            requires_grad=True,
+        ),
+    ),
+    (-0.1111, -0.2222),
+    (-0.1111, -0.2222, 0.4554),
+    (-0.1735, -0.1735, -0.2846, -0.2846),
+    (-0.1735, -0.2846),
+    (-0.1735, -0.2846),
+    (-0.1735,),
+    (-0.1111, 0.3333),
+]
+
+
+def autodiff_metric_tensor(ansatz, num_wires):
+    """Compute the metric tensor by full state vector
+    differentiation via autograd."""
+    dev = qml.device("default.qubit", wires=num_wires)
+
+    @qml.qnode(dev)
+    def qnode(*params):
+        ansatz(*params, wires=dev.wires)
+        return qml.state()
+
+    def mt(*params):
+        state = qnode(*params)
+        rqnode = lambda *params: np.real(qnode(*params))
+        iqnode = lambda *params: np.imag(qnode(*params))
+        jac = qml.jacobian(rqnode)(*params) + 1j * qml.jacobian(iqnode)(*params)
+        psidpsi = np.tensordot(np.conj(state), jac, axes=([0], [0]))
+        return np.real(
+            np.tensordot(np.conj(jac), jac, axes=([0], [0]))
+            - np.tensordot(np.conj(psidpsi), psidpsi, axes=0)
+        )
+
+    return mt
+
+
+class TestFullMetricTensor:
+
+    num_wires = 3
+
+    @pytest.mark.parametrize("ansatz, params", zip(fubini_ansatze, fubini_params))
+    def test_correct_output_autograd(self, ansatz, params):
+        expected = autodiff_metric_tensor(ansatz, self.num_wires)(*params)
+        dev = qml.device("default.qubit.autograd", wires=self.num_wires + 1)
+
+        @qml.qnode(dev, interface="autograd")
+        def circuit(*params):
+            """Circuit with dummy output to create a QNode."""
+            ansatz(*params, dev.wires[:-1])
+            return qml.expval(qml.PauliZ(0))
+
+        mt = qml.metric_tensor(circuit, approx=None)(*params)
+
+        assert np.allclose(mt, expected)
+
+    @pytest.mark.skip(reason="JAX does not support the forward pass metric tensor.")

```

Comment:
I was thinking about this a bit more - I think we could solve this problem by having
```python
qml.metric_tensor(qnode, argnums=..)(*args)
```

We could have ~the following logic:

```python
argnums = qml.math.get_trainable_indices(args)

if not argnums:
    # assume *all* positional arguments are trainable
```

Reply:
That's a very good solution!!

Reply:
Opened an issue: #1880